src/macaron/slsa_analyzer/analyzer.py

```diff
@@ -433,6 +433,41 @@ def run_single(
         analyze_ctx.dynamic_data["expectation"] = self.expectations.get_expectation_for_target(
             analyze_ctx.component.purl.split("@")[0]
         )
+
+        git_service = self._determine_git_service(analyze_ctx)
+        self._determine_ci_services(analyze_ctx, git_service)
+        self._determine_build_tools(analyze_ctx, git_service)
+        self._determine_package_registries(analyze_ctx)
```

Comment:
I am curious whether these operations are any different from what we have in https://github.com/oracle/macaron/blob/927d8c3a2e59b0b323036ff5bd309fa765248899/src/macaron/slsa_analyzer/analyzer.py#L1121

From what I see, `perform_checks()` is called at https://github.com/oracle/macaron/blob/927d8c3a2e59b0b323036ff5bd309fa765248899/src/macaron/slsa_analyzer/analyzer.py#L478
and still contains the logic for findings CI services, Git service, etc.

Reply:
Indeed they are the same and `perform_checks` should be reduced/removed.